src/app/products/charges/create-charge/create-charge.component.html

```diff
@@ -0,0 +1,200 @@
+              <mat-label>Due Date</mat-label>
+              <input matInput [min]="minDate" [max]="maxDate" [matDatepicker]="dueDatePicker" required formControlName="feeOnMonthDay">
+              <mat-datepicker-toggle matSuffix [for]="dueDatePicker"></mat-datepicker-toggle>
+              <mat-datepicker #dueDatePicker></mat-datepicker>
+              <mat-error *ngIf="chargeForm.controls.feeOnMonthDay.hasError('required')">
+                Due Date is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <div *ngIf="chargeForm.controls.chargeTimeType.value === 7 || chargeForm.controls.chargeTimeType.value === 11" fxFlex="48%" fxLayout="row">
+              <mat-form-field fxFlex="80%">
+                <mat-label>Repeat Every</mat-label>
+                <input fxFlex="80%" matInput required formControlName="feeInterval">
+                <mat-error *ngIf="chargeForm.controls.feeInterval.hasError('required')">
+                  Repeat Every is <strong>required</strong>
+                </mat-error>
+                <mat-error *ngIf="chargeForm.controls.feeInterval.hasError('min') || chargeForm.controls.feeInterval.hasError('max')">
+                  Repeat Every <strong>must be between 1 and 12 months</strong>
+                </mat-error>
+                <mat-error *ngIf="chargeForm.controls.feeInterval.hasError('pattern')">
+                  Repeat Every <strong>must be an integer</strong>
+                </mat-error>
+              </mat-form-field>
+
+              <div fxFlex="20%" class="repeat-every-label-wrapper">
+                <mat-label class="repeat-every-label mat-h4">{{ repeatEveryLabel }}</mat-label>
+              </div>
+            </div>    
+  
+            <mat-form-field fxFlex="48%">
+              <mat-label>Amount</mat-label>
+              <input matInput required formControlName="amount">
+              <mat-error *ngIf="chargeForm.controls.amount.hasError('required')">
+                Amount is <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="chargeForm.controls.amount.hasError('pattern')">
+                Amount must be a <strong>number larger than 0</strong>
+              </mat-error>
+            </mat-form-field>
+            
+            <div fxFlex="48%" fxLayout="row" fxLayoutGap="2%" fxLayout.lt-md="column">
+              <div fxFlex="50%" class="active-wrapper">
+                <mat-checkbox labelPosition="before" formControlName="active">
+                  Active
+                </mat-checkbox>
+              </div>
+  
+              <div fxFlex="50%" class="penalty-wrapper">
+                <mat-checkbox labelPosition="before" formControlName="penalty" [disabled]="chargeForm.controls.chargeAppliesTo.value === 4 || chargeForm.controls.chargeTimeType.value === 9">
+                  Is Penalty
+                </mat-checkbox>
```

Comment:
Its recommended to set the disabled attribute from the form instead: (See below screenshot of the console for the warning). Make the changes accordingly.
![C1](https://user-images.githubusercontent.com/16948598/71926300-0cf4f880-31a4-11ea-9bd1-04fc8cc35751.PNG)

Reply:
I have been trying to do it the way that you mean (through the form) for hours already and I simply could not be able to get it to work. Setting it the disabled value in the form means that the value is not included in the form (setting it in html includes it), while a readonly property simply does not exist in angular mat-checkboxes.

Reply:
Ignore what I have just said, I have managed to find a way to bypass all this :)